### PR TITLE
Add is:unlisted tag to discontinued/legacy packages.

### PR DESCRIPTION
**Pull request by isoos**
Required for #5988.

### CODE REVIEW

**Comment by jonasfj** [comment]
@isoos can't we repurpose the `is:unlisted` tag?

**Comment by jonasfj** [comment]
We could also do `is:unlisted OR is:discontinued`.

But I think it might be okay to say that discontinued packages are unlisted.

**Comment by jonasfj** [comment]
Actually, in the admin UI, you can't set `unlisted` if you've set `discontinue`. So I think it's fair if we argue that `discontinued` is a subsets of `unlisted`.

**Comment by jonasfj** [comment]
Then we should probably also tweak the search filter is `include unlisted` in the advanced section.

**Comment by isoos** [comment]
> Actually, in the admin UI, you can't set unlisted if you've set discontinue. So I think it's fair if we argue that discontinued is a subsets of unlisted.

But you can set `unlisted` without setting it discontinued, so repurposing it will lose some logical correctness.

> Then we should probably also tweak the search filter is include unlisted in the advanced section.

The current `show:hidden` enables both discontinued, unlisted and legacy (Dart 1.x). I think we should have a filter that enables "search everything" in one way or another.

**Comment by jonasfj** [comment]
Maybe, we should update the UI for setting something as "discontinued" to say "set discontinue and unlisted", or something like that?

Maybe just in the explanatory text, such that people know that a discontinued package is always also unlisted.

Won't that remove the cognitive difference, and make `is:unlisted` a subsets of packages are that discontinued or unlisted?


-----------------

Context: I dislike the word "hidden", it gives connotations that makes a user think they are not visible to others, which is just not true. So using this word can cause surprises -- which in this context is a privacy issue.

**Comment by isoos** [comment]
Refactored with the intent that we will change "Include hidden" to "Include unlisted".
PTAL.

File: pkg/_pub_shared/lib/search/tags.dart [hunk 1]
```diff
@@ -19,9 +19,6 @@ const allowedTagPrefixes = [
 
 /// Collection of package-related tags.
 abstract class PackageTags {
-  /// Package is marked discontinued | unlisted | legacy.
-  static const String isHidden = 'is:hidden';
```
Comment (jonasfj) [inline pkg/_pub_shared/lib/search/tags.dart:23]: nit: Need to update the comment for `isUnlisted`.

**Review by jonasfj — APPROVED** [review]

**Pull request merged**